src/test/java/com/teragrep/cfe_39/HdfsTest.java

```diff
@@ -0,0 +1,260 @@
+                SyslogRecord lastRecord = null;
+                try (DataFileReader<SyslogRecord> dataFileReader = new DataFileReader<>(syslogFile, userDatumReader)) {
+                    while (dataFileReader.hasNext()) {
+                        lastRecord = dataFileReader.next(lastRecord);
+                    }
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+
+                assert lastRecord != null;
+                RecordOffset lastObject = new RecordOffset("testConsumerTopic", Integer.parseInt(lastRecord.getPartition().toString()), lastRecord.getOffset(), null); // Fetch input parameters from the lastRecord SyslogRecord-object.
+                LOGGER.debug("\n"+"Last record in the " + syslogFile.getName() + " file:" + "\ntopic: " + lastObject.getTopic() + "\npartition: " + lastObject.getPartition() + "\noffset: " + lastObject.getOffset());
+                try (HDFSWrite writer = new HDFSWrite(config, lastObject)) {
+                    writer.commit(syslogFile, -1L); // commits the final AVRO-file to HDFS.
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+
+                // Check that the file was stored to HDFS properly.
+                try {
+                    Thread.sleep(1000);
+                    hdfsReadCheck("testConsumerTopic", Integer.parseInt(lastRecord.getPartition().toString()), lastRecord.getOffset());
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+
+            }
+        }
+        closeMiniCluster();
+    }
+
+    public void hdfsReadCheck(String testConsumerTopic, int partition, long offset) throws IOException {
+
+        // Check that the files were properly written to HDFS with a read test.
+        String hdfsuri = config.getHdfsuri();
+
+        String path = config.getHdfsPath()+"/"+testConsumerTopic;
+        String fileName = partition+"."+offset;
+        // ====== Init HDFS File System Object
+        Configuration conf = new Configuration();
+        // Set FileSystem URI
+        conf.set("fs.defaultFS", hdfsuri);
+        // Because of Maven
+        conf.set("fs.hdfs.impl", org.apache.hadoop.hdfs.DistributedFileSystem.class.getName());
+        conf.set("fs.file.impl", org.apache.hadoop.fs.LocalFileSystem.class.getName());
+        // Set HADOOP user
+        System.setProperty("HADOOP_USER_NAME", "hdfs");
+        System.setProperty("hadoop.home.dir", "/");
+        //Get the filesystem - HDFS
+        FileSystem fs = FileSystem.get(URI.create(hdfsuri), conf);
+
+        //==== Create directory if not exists
+        Path workingDir=fs.getWorkingDirectory();
+        Path newDirectoryPath= new Path(path);
+        if(!fs.exists(newDirectoryPath)) {
```

Comment:
No conditionals in tests

Reply:
Solved in commit 65342c2e865912f981dfe82f470cab6884f72cdb